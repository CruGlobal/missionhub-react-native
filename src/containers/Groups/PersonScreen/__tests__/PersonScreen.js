import React from 'react';

import {
  PersonScreen,
  ContactPersonScreen,
  MemberPersonScreen,
} from '../PersonScreen';
import {
  renderShallow,
  createMockStore,
  testSnapshotShallow,
  createMockNavState,
} from '../../../../../testUtils';

const store = createMockStore({});

const organization = { id: '1', name: 'Test Org' };
const person = { id: '1', full_name: 'Test Person' };

describe('Contact', () => {
  it('should render PersonScreen correctly', () => {
    testSnapshotShallow(
      <PersonScreen
        navigation={createMockNavState({
          organization,
          person,
        })}
      />,
      store,
    );
  });

  it('should render ContactPersonScreen correctly', () => {
    testSnapshotShallow(
      <ContactPersonScreen
        navigation={createMockNavState({
          organization,
          person,
        })}
      />,
      store,
    );
  });

  it('should render MemberPersonScreen correctly', () => {
    testSnapshotShallow(
      <MemberPersonScreen
        navigation={createMockNavState({
          organization,
          person,
        })}
      />,
      store,
    );
  });
});
