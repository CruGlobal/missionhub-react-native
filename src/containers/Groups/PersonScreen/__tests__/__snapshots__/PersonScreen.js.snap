// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact should render PersonScreen correctly 1`] = `
<Component>
  <Connect(Header)
    left={<Connect(BackButton) />}
    right={
      <IconButton
        name="moreIcon"
        onPress={[Function]}
        type="MissionHub"
      />
    }
    shadow={false}
    title="Test Org"
  />
  <Flex
    align="center"
    justify="center"
    self="stretch"
    style={
      Object {
        "backgroundColor": "#007398",
        "paddingVertical": 10,
      }
    }
  >
    <MyText
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 24,
        }
      }
    />
  </Flex>
</Component>
`;
